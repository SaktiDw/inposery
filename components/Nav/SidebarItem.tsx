import Link from "next/link";
import React from "react";

type Props = {
  href: string;
  active: boolean;
  icon: string;
  text: string;
};

const SidebarItem = (props: Props) => {
  return (
    <Link
      href={props.href}
      className={`p-2 rounded-lg flex outline-none
        ${
          props.active &&
          "bg-white shadow-lg dark:bg-slate-800 dark:text-slate-100"
        } group hover:bg-white hover:text-slate-800 hover:shadow-lg hover:scale-105 transition-all ease-in-out duration-200
        dark:hover:bg-slate-800 dark:hover:text-slate-100`}
    >
      <>
        <i
          className={`w-10 h-10 rounded-lg flex justify-center items-center transition-all ease-in-out duration-200 shadow group-hover:text-white group-hover:bg-gradient-to-tl from-green-700 to-lime-500  ${
            props.active
              ? "bg-gradient-to-tl from-green-700 to-lime-500 text-white"
              : "bg-white shadow-xl text-slate-800 dark:bg-slate-800 dark:text-slate-100"
          } ${props.icon}`}
        ></i>
        <span className="p-2">{props.text}</span>
      </>
    </Link>
  );
};

export default SidebarItem;
