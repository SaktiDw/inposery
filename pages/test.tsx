import { StoreDashboard } from "@/components";
import axiosInstance from "@/helper/lib/client";
import React from "react";
import useSWR from "swr";

type Props = {};

const Test = (props: Props) => {
  const { data: stores } = useSWR("/api/stores", async (url: string) => {
    await axiosInstance.get(url).then((res) => res.data);
  });
  return <StoreDashboard>{JSON.stringify(stores)}</StoreDashboard>;
};

export default Test;
