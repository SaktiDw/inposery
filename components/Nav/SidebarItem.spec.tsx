import { bench, describe, expect, test, vi } from "vitest";
import { render, screen, within } from "@testing-library/react";
import SidebarItem from "./SidebarItem";
import Home from "@/pages/index";
import { mockNextUseRouter } from "@/helper/test/utils";

describe("SidebarItem", async () => {
  test("Render SidebarItem", () => {
    mockNextUseRouter({
      route: "/",
      pathname: "/",
      query: "",
      asPath: "/",
    });

    const { container } = render(
      <SidebarItem
        href="/test"
        text="Test"
        active={true}
        icon="fi-rr-test-tube"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
