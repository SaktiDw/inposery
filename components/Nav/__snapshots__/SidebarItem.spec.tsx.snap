// Vitest Snapshot v1

exports[`SidebarItem > Render SidebarItem 1`] = `
<div>
  <i
    class="w-10 h-10 rounded-lg flex justify-center items-center transition-all ease-in-out duration-200 shadow group-hover:text-white group-hover:bg-gradient-to-tl from-green-700 to-lime-500  bg-gradient-to-tl from-green-700 to-lime-500 text-white fi-rr-test-tube"
  />
  <span
    class="p-2"
  >
    Test
  </span>
</div>
`;
